app/tests/test_generic_api.py

```diff
@@ -293,3 +296,79 @@ def test_delete_precondition(self):
+        cls.collection = cls.factory.create_collection_sample(name='collection-1').model
+        cls.item = cls.factory.create_item_sample(collection=cls.collection, name='item-1').model
+        cls.asset = cls.factory.create_asset_sample(item=cls.item, name='asset-1').model
+
+    @override_settings(CACHE_MIDDLEWARE_SECONDS=3600)
+    def test_get_cache_header(self):
+        for endpoint in [
+            '',  # landing page
+            'conformance',
+            'search',
+            'search?ids=item-1',
+            'collections/collection-1',
+            'collections/collection-1/items/item-1',
+            'collections/collection-1/items/item-1/assets/asset-1'
+        ]:
+            with self.subTest(endpoint=endpoint):
+                now = datetime.now()
+                response = self.client.get(f"/{API_BASE}/{endpoint}")
+                self.assertStatusCode(200, response)
+
+                self.assertTrue(
+                    response.has_header('Cache-Control'), msg="Cache-Control header missing"
+                )
+                self.assertEqual(
+                    response['Cache-Control'],
+                    'max-age=3600, public',
+                    msg='Wrong cache-control values'
+                )
+
+                self.assertTrue(response.has_header('Expires'), msg="Expires header missing")
+
+                expires = datetime.strptime(response['Expires'], '%a, %d %b %Y %H:%M:%S GMT')
+                self.assertAlmostEqual((expires - now).total_seconds(),
+                                       timedelta(seconds=3600).total_seconds(),
+                                       delta=2)
+
+    @override_settings(CACHE_MIDDLEWARE_SECONDS=3600)
+    def test_head_cache_header(self):
+        for endpoint in [
+            '',  # landing page
+            'conformance',
+            'search',
+            'search?ids=item-1',
+            'collections/collection-1',
+            'collections/collection-1/items/item-1',
+            'collections/collection-1/items/item-1/assets/asset-1'
+        ]:
+            with self.subTest(endpoint=endpoint):
+                now = datetime.now()
+                response = self.client.head(f"/{API_BASE}/{endpoint}")
+                self.assertStatusCode(200, response)
+
+                self.assertTrue(
+                    response.has_header('Cache-Control'), msg="Cache-Control header missing"
+                )
+                self.assertEqual(
+                    response['Cache-Control'],
+                    'max-age=3600, public',
+                    msg='Wrong cache-control values'
+                )
+
+                self.assertTrue(response.has_header('Expires'), msg="Expires header missing")
+
+                expires = datetime.strptime(response['Expires'], '%a, %d %b %Y %H:%M:%S GMT')
+                self.assertAlmostEqual((expires - now).total_seconds(),
+                                       timedelta(seconds=3600).total_seconds(),
```

Comment:
Nice, thanks for adding the unit tests.
Just have to small points here:

* Probably the 3600 should be a variable and also be uses in the `override_settings` decorator, so in case the value will be changed in future in one place, it will be changed everywhere and the test will not crash
* It is probably very unlikely that a delta of 2 is not enough, but would there be an alternative to using `AlmostEqual` here? Because what we want to check here is, that the `Expires` is something between 3600 and 0 (or simply `le` 3600, right?) and not that this check can be done in less than 2 seconds after the request was send.

Reply:
Resolved :-)